src/components/ConfigEditor/AzureCredentialsConfig.ts

```diff
@@ -0,0 +1,207 @@
+import { DataSourceSettings } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+import { AzureCloud, AzureCredentials, ConcealedSecret } from './AzureCredentials';
+
+const concealed: ConcealedSecret = Symbol('Concealed client secret');
+const concealedLegacy: ConcealedSecret = Symbol('Concealed legacy client secret');
```

Comment:
is there any benefit of doing this distinction? AFAIK the logic will be the same.

Reply:
`clientSecret` of `AzureClientSecretCredentials` has type `string | symbol`.

the `string` is when user changed the secret and actual secret stored in the string. the `symbol` is when we know that secret is stored on backend but frontend doesn't have its actual value.

For outside code its only either `string` or `symbol`. The `concealed` and `concealedLegacy` variables are not exported outside the module and used internally to handle the legacy scenario.

Newly created credentials store secret in the `azureClientSecret` field of `SecureJsonData`, legacy credentials store in `clientSecret` field. When `getCredentials` function reads credentials from JsonData/SecureJsonData, it returns `AzureCredentials` object and information about which field was used is being lost.

When `updateCredentials` method writes the `AzureCredentials` object back to JsonData/SecureJsonData and secret wasn't changed, then it should preserve the whatever the original field was used. So difference between `concealed` and `concealedLegacy` is information that tells the `updateCredentials` method whether the  `azureClientSecret` or `clientSecret` field was originally used.

This information known only to the module, for external code it's just `symbol` with no any difference.